client/blocks/signup-form/index.jsx

```diff
@@ -233,7 +235,22 @@ class SignupForm extends Component {
 			this.props.displayNameInput && 'lastName',
 		] );
 
+		// Object with all fields
 		const data = mapKeys( pick( fields, fieldsForValidation ), ( value, key ) => snakeCase( key ) );
+
+		// Remove fields that have yet to be visited
+		forEach( data, ( value, field ) => {
+			// prevent an error messages from being shown prematurely
+			if ( field === 'password' && ! this.state.focusPassword ) {
+				delete data.password;
+			}
+
+			if ( field === 'username' && ! this.state.focusUsername ) {
+				delete data.username;
+			}
```

Comment:
What if you excluded the unwanted fields when creating the `fieldsForValidation` array?
```js
const fieldsForValidation = filter( [
  'email',
  this.state.focusPassword && 'password',
  this.props.displayUsernameInput && this.state.focusUsername && 'username',
  this.props.displayNameInput && 'firstName',
  this.props.displayNameInput && 'lastName',
] );
```

Reply:
Even better! Updated in https://github.com/Automattic/wp-calypso/commit/0c152b503e061e065b6d320e052bea0add56b3c0 Thanks for the improvements @jsnajdr!